src/Widgets/DiskBar.vala

```diff
@@ -24,113 +24,88 @@ public class Installer.DiskBar: Gtk.Grid {
-        bar_style_context.add_class ("disk-bar");
-
-        bar.size_allocate.connect ((alloc) => {
-            update_sector_lengths (partitions, alloc);
-        });
+        show_all ();
+    }
 
-        foreach (PartitionBar part in partitions) {
-            bar.add (part);
+    private uint64 get_unused () {
+        uint64 used = 0;
+        foreach (PartitionBar partition in partitions) {
+            used += partition.get_size ();
         }
 
-        legend_container = new Gtk.Grid ();
-        legend_container.column_spacing = 12;
-        legend_container.halign = Gtk.Align.CENTER;
-        legend_container.margin_bottom = 9;
+        return size - (used * 512);
+    }
 
-        var legend = new Gtk.ScrolledWindow (null, null);
+    private void generate_legend () {
+        legend = new Gtk.ScrolledWindow (null, null);
         legend.vscrollbar_policy = Gtk.PolicyType.NEVER;
+
+        legend_container = new Gtk.Box (Gtk.Orientation.HORIZONTAL, 12);
+        legend_container.set_halign (Gtk.Align.CENTER);
+        legend_container.margin_bottom = 9;
         legend.add (legend_container);
 
         foreach (PartitionBar p in partitions) {
-            add_legend (p.path, p.get_size () * 512, Distinst.strfilesys (p.filesystem), p.vg, p.menu);
-        }
-
-        uint64 used = 0;
-        foreach (PartitionBar partition in partitions) {
-            used += partition.get_size ();
+            add_legend (p.path, p.get_size() * 512, Distinst.strfilesys (p.filesystem), p.vg, p.menu);
         }
 
-        var unused = size - (used * 512);
         if (size / 100 < unused) {
             add_legend ("unused", unused, "unused", null, null);
-
-            var unused_bar = new Gtk.Grid ();
-            unused_bar.expand = true;
-
-            var unused_style_context = unused_bar.get_style_context ();
-            unused_style_context.add_class ("fill-block");
-            unused_style_context.add_class ("unused");
-
-            bar.add (unused_bar);
         }
-
-        hexpand = true;
-        margin = 6;
-        get_style_context ().add_class (Granite.STYLE_CLASS_STORAGEBAR);
-        attach (label, 0, 1);
-        attach (legend, 1, 0);
-        attach (bar, 1, 1);
-
-        show_all ();
     }
 
     private void add_legend (string ppath, uint64 size, string fs, string? vg, Gtk.Popover? menu) {
         var fill_round = new FillRound ();
         fill_round.width_request = fill_round.height_request = 14;
-        fill_round.valign = Gtk.Align.CENTER;
+        fill_round.set_valign(Gtk.Align.CENTER);
 
         var context = fill_round.get_style_context ();
         context.add_class ("legend");
         context.add_class (fs);
 
         var format_size = GLib.format_size (size);
-
         var info = new Gtk.Label (
             (vg == null)
-                ? _("%s (%s)").printf (format_size, fs)
-                : _("%s (%s: <b>%s</b>)").printf (format_size, fs, vg)
+                ? "%s (%s)".printf (format_size, fs)
```

Comment:
I believe these should still be translatable.

Reply:
I don't see how, since `%s` doesn't translate.

Reply:
The ordering of the strings can be region-specific, as can the use of parens versus other characters.